tensorboard/webapp/widgets/linked_time_fob/linked_time_types.ts

```diff
@@ -19,3 +19,49 @@ export interface LinkedTime {
   start: {step: number};
   end: {step: number} | null;
 }
+
+/**
+ * This class is intended to be implemented by the card that has a
+ * LinkedTimeFobControllerComponent.
+ *
+ * Each implementer will have some sort of Scale that is used to translate
+ * convert between step and pixel so that the fob lines up with the axis
+ * properly. In future comments this scale will be refered to as
+ * ImplementerScale
+ *
+ * It also allows cards to implement the dragging functionality in different
+ * ways. By deciding which step to return in the getStepHigherThanMousePosition
+ * and getStepLowerThanMousePosition functions the implementer can decide if the
+ * fob "snaps" to certain steps or drags in in a smooth continuous way.
+ */
+export interface FobCardAdapter {
+  /**
+   * gets the highest step for this card.
+   */
+  getHighestStep(): number;
+
+  /**
+   * Gets the lowest step for this card.
+   */
+  getLowestStep(): number;
+
+  /**
+   * Uses ImplementerScale to translate the proper pixel offset.
+   * @param step the step which needs to be translated.
+   */
+  stepToPixel(step: number): number;
```

Comment:
is stepToPixel sort of the inverse operation of getStepHigherThanMousePosition and getStepLowerThanMousePosition? Maybe not strictly, mathematically, but in spirit? If so, would it make sense to try to use similar terms in the function names to reflect this?

Maybe:
getFobPositionFromStep()
getStepHigherThanMousePosition()
getStepLowerThanMousePosition()
(Also note that I have modified the ordering of terms for consistency - the functions are either "getting A from B" or "getting B from A", rather than having one function that is "A to B" and others that are "A from B".)

Or if we want to make the terminology even more generic and more clearly related, something like:
getAxisPositionFromStep()
getStepHigherThanAxisPosition()
getStepLowerThanAxisPosition()

Reply:
I like these suggestions. I am going with the AxisPosition names.